src/spring-cloud/azext_spring_cloud/_params.py

```diff
@@ -147,9 +147,13 @@ def prepare_logs_argument(c):
     for scope in ['spring-cloud app deploy', 'spring-cloud app deployment create']:
         with self.argument_context(scope) as c:
             c.argument(
-                'artifact_path', options_list=['--artifact-path', '--jar-path', '-p'],
-                help='If provided, deploy pre-built artifact (jar or netcore zip), otherwise deploy current folder as tar.',
-                validator=validate_jar)
+                'artifact_path', options_list=['--artifact-path',
+                                               c.deprecate(target='--jar-path', redirect='--artifact-path', hide=True),
+                                               c.deprecate(target='-p', redirect='--artifact-path', hide=True)],
```

Comment:
May I ask why do you need to deprecate those options?

Reply:
'--jar-path' and '-p' are the old ways for user to provide the java artifact. Now we support .net artifact so '--jar-path' is not a accurate argument. Also, we support '--source-path' so '-p' may make users confused. That is why we need to deprecate these two options.

Reply:
Got it, thanks~